### ISSUE TITLE
Consider handling falsey values i.e. parse('' | undefined | null)

### ISSUE THREAD

**Issue by clodal** [beenotung/best-effort-json-parser#1]
Consider handling empty string by default or as an option to avoid `SyntaxError: Unexpected end of JSON input`.

## Issue

We should handle falsey values by default:
```
// run.js
import { parse } from 'best-effort-json-parser'

parse('') // Actual: Unhandled Runtime Error. SyntaxError: Unexpected end of JSON input. Expected: {}
parse(undefined) // Actual: TypeError: Cannot read properties of undefined (reading '0'). Expected: {}
parse(null) // Actual: returns `null`. Expected: {}
```

Alternative is to handle this as an option e.g. 
```
// RFC
parse(null, { disableFallback/enableFallback: true })
```

## Benefits
Downstream usage of this function would be able to get assurance that the function will always return a JS plain object so that it can reliably manage its type.

```
const completion: string = await fetchStreamingCompletion()
const myStreamingObject = parse(completion) // Assuming initial `completion` value is an empty string: ''.
console.log(myStringObject.hello) // Actual: Unhandled Runtime Error. SyntaxError: Unexpected end of JSON input. Expected: undefined
```

Thoughts? @beenotung

**Comment by beenotung** [comment]
Would it be more flexible to check for empty string before calling the parse function?

Since different use cases may prefer to treat empty string as different value, could be empty string, zero number, empty object or empty array.

**Comment by beenotung** [comment]
The update is released to npm as `best-effort-json-parser@1.1.0` :rocket: 

Feel free to re-open the issue if it doesn't address your case
